src/Runtime/WideChar.cs

```diff
@@ -0,0 +1,93 @@
+namespace CppSharp.Runtime
+{
+    // Struct added to help map a the wchar_t C++ type to C#
+    // The data is stored as a 64 bit value so it could represent
+    // an UTF-32 character but C# only represents UTF-16 characters
+    // so beware of possible data loss when using it.
```

Comment:
Not sure what this means, is it that our binding for wstring does not always work properly?

Reply:
Just that C# char (UTF-16) type cannot represent the entirety of Unicode (UTF-32).

Reply:
We should then think about throwing an exception if out of range.

Reply:
I don't get this comment. `wchar_t` on Windows is represented as UTF16-LE. On Linux it's UTF-32. Char in C# is always UTF16-LE afaik. There is one more type to consider: `char16_t` is always UTF16-LE, regardless of the platform that's being used. (Since C++17). C++20 will also define UTF8 type `char8_t`.